cpp/include/cudf/strings/string_view.inl

```diff
@@ -84,6 +88,16 @@ __device__ inline size_type string_view::length() const
 {
     if( _length <= UNKNOWN_STRING_LENGTH )
         _length = strings::detail::characters_in_string(_data,_bytes);
+    if( _length && (_char_width <= UNKNOWN_CHAR_WIDTH) )
+    {
+        const BYTE* bytes = reinterpret_cast<const BYTE*>(data());
+        auto chwidth = bytes_in_utf8_byte(*bytes); // see if they are all the same width
+        _char_width = (thrust::find_if( thrust::seq, bytes, bytes + size_bytes(), 
+                            [chwidth] (auto ch) {
+                                auto width = bytes_in_utf8_byte(ch);
+                                return (width!=0) && (width!=chwidth);
+                            }))==(bytes + size_bytes()) ? chwidth : 0;
```

Comment:
Looks like you are using `_char_width == 0`  to mean  "variable character  width".  In  that  case shouldn't you  have a 

```
constexpr int8_t VARIABLE_CHAR_WIDTH{0};
```

Rather than using literal zero?